src/main/java/com/mraof/minestuck/event/ServerEventHandler.java

```diff
@@ -260,11 +261,40 @@ public static void onPlayerTickEvent(TickEvent.PlayerTickEvent event)
 		if(!event.player.level.isClientSide)
 		{
 			PlayerData data = PlayerSavedData.getData((ServerPlayer) event.player);
+			Player player = event.player;
+			
 			if(data.getTitle() != null)
+			{
 				data.getTitle().handleAspectEffects((ServerPlayer) event.player);
+				IdentifierHandler.encode(player);
+			}
+			
+			Level level = player.level;
+			SessionHandler sessionHandler = SessionHandler.get(level);
+			Session session = sessionHandler.getPlayerSession(IdentifierHandler.encode(player));//finds the session they're in
+			long gameTime = level.getGameTime();
+			int inputTime = 200;
+			
+			NonNegativeGristSet playerCache = new NonNegativeGristSet(PlayerSavedData.getData((ServerPlayer) player).getGristCache());
+			if(session == null)
+			{
+				return;
+			}
+			if(gameTime % inputTime == 0)
+			{//every tick, the server will try to put some of the gutter's grist into each player's cache
+				int gutterMultiplier = (int) session.getGutterMultiplier();
+				int rung = PlayerSavedData.getData((ServerPlayer) player).getEcheladder().getRung();//finds the target's rung
+				int spliceAmount = (int) (GristHelper.rungGrist[rung] * Math.min((gutterMultiplier + 1.0), 1.0) / 20.0);//determines the appropriate splice amount
+				
+				GristGutter sessionGutter = session.getGristGutter();//get's the grist gutter
+				playerCache.addGrist(sessionGutter.splice(spliceAmount));//splices some grist from the splice set
+				GristSet rungGrist = GristHelper.limitGristByPlayerRung(level, IdentifierHandler.encode(player), playerCache);
+				
+				sessionGutter.addGrist(rungGrist);
+				data.setGristCache(playerCache);
+			}
```

Comment:
Would have to check with Kirderf to see if there is a better solution and real my concerns actually are, but it sounds resource intensive to have this kind of data looked up every tick

Reply:
understood